src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/formats/html/HtmlIds.java

```diff
@@ -595,4 +595,33 @@ public HtmlId forHeading(CharSequence headingText, Set<String> headingIds) {
         }
         return HtmlId.of(idValue);
     }
+
+    /**
+     * Returns an id for a snippet.
+     *
+     * @param e the element in whose documentation the snippet appears
+     * @param snippetIds the set of snippet ids already generated
+     * @return a unique id for the snippet
+     */
+    public HtmlId forSnippet(Element e, Set<String> snippetIds) {
+        String id = "snippet-";
+        ElementKind kind = e.getKind();
+        if (kind == ElementKind.PACKAGE) {
+            id += forPackage((PackageElement) e).name();
+        } else if (kind.isClass()) {
+            id += forClass((TypeElement) e).name();
+        } else if (kind.isExecutable()) {
+            id += forMember((ExecutableElement) e).getFirst().name();
+        } else if (kind.isField()) {
+            id += forMember((VariableElement) e).name();
+        } else {
+            // utmost unexpected, but this case shouldn't fail
+            id += "unknown-element-";
```

Comment:
Feels weird that unknown element has a `-` between name and count but the others don't.

Reply:
Sorry, I initially pushed an outdated commit. This is a rare situation where a force push is warranted. Also, fixed.